src/test/java/edu/gonzaga/CityTest.java

```diff
@@ -0,0 +1,285 @@
+
+        City testCity1 = new City(Color.BLUE, "Los Angeles", true);
+        City testCity2 = new City(Color.BLACK, "San Francisco");
+
+
+        Assertions.assertEquals(desiredValue, testCity2.findConnection(testCity1));
+    }
+
+    /**
+     * Tests that a city can have cubes added
+     * 
+     * @author Aiden T
+     */
+    @Test
+    void cityInfection1() {
+        ArrayList<Color> desiredValue = new ArrayList<Color>();
+        desiredValue.add(Color.BLACK);
+
+        City testCity1 = new City(Color.BLUE, "Los Angeles", true);
+        testCity1.addInfectionCube(Color.BLACK);
+
+        Assertions.assertEquals(desiredValue, testCity1.getInfectionCubes());
+    }
+
+    /**
+     * Tests that a city can have cubes of multiple colors added
+     * 
+     * @author Aiden T
+     */
+    @Test
+    void cityInfection2() {
+        ArrayList<Color> desiredValue = new ArrayList<Color>();
+        desiredValue.add(Color.BLUE);
+        desiredValue.add(Color.BLUE);
+        desiredValue.add(Color.BLACK);
+
+        City testCity1 = new City(Color.BLUE, "Los Angeles", true);
+        testCity1.addInfectionCube(Color.BLUE);
+        testCity1.addInfectionCube(Color.BLUE);
+        testCity1.addInfectionCube(Color.BLACK);
+
+        Assertions.assertEquals(desiredValue, testCity1.getInfectionCubes());
+    }
+
+    /**
+     * Tests that a city can remove cubes
+     * 
+     * @author Aiden T
+     */
+    @Test
+    void cityRemoveInfection1() {
+        ArrayList<Color> desiredValue = new ArrayList<Color>();
+        desiredValue.add(Color.BLUE);
+
+        City testCity1 = new City(Color.BLUE, "Los Angeles", true);
+        testCity1.addInfectionCube(Color.BLUE);
+        testCity1.addInfectionCube(Color.BLUE);
+        testCity1.removeInfectionCube(Color.BLUE);
+
+        Assertions.assertEquals(desiredValue, testCity1.getInfectionCubes());
+    }
+
+    /**
+     * Tests that a city can remove cubes of multiple colors
+     * 
+     * @author Aiden T
+     */
+    @Test
+    void cityRemoveInfection2() {
+        ArrayList<Color> desiredValue = new ArrayList<Color>();
+        desiredValue.add(Color.BLUE);
+        desiredValue.add(Color.BLUE);
+
+        City testCity1 = new City(Color.BLUE, "Los Angeles", true);
+        testCity1.addInfectionCube(Color.BLUE);
+        testCity1.addInfectionCube(Color.BLUE);
+        testCity1.removeInfectionCube(Color.BLACK);
+
+        Assertions.assertEquals(desiredValue, testCity1.getInfectionCubes());
+    }
+
+    /**
+     * Tests that a city can remove a research station
+     * 
+     * @author Aiden T
+     */
+    @Test
+    void cityResearchStation1() {
+        Boolean desiredValue = false;
+
+        City testCity1 = new City(Color.BLUE, "Los Angeles", true);
+        testCity1.removeResearchStation();
+
+        Assertions.assertEquals(desiredValue, testCity1.getResearchStation());
+    }
+
+    /**
+     * Tests that a city can't double add a research station
```

Comment:
Same as earlier comment, test requires reading console to make sure function works